### PR TITLE
Introduce `FluxFromStreamSupplier` Refaster rule

### PR DESCRIPTION
**Pull request by rickie**

### CODE REVIEW

File: error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java [hunk 1]
```diff
@@ -1947,4 +1948,20 @@ Mono<T> after(CompletableFuture<T> future, boolean suppressCancel) {
       return Mono.fromFuture(() -> future, suppressCancel);
     }
   }
+
+  /**
+   * Prefer {@link Flux#fromStream(Supplier)} over {@link Flux#fromStream(Stream)}, as the former
+   * may defer initiation of the asynchronous computation until subscription.
+   */
+  static final class FluxFromStreamSupplier<T> {
+    @BeforeTemplate
```
Comment (EnricSala) [inline error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java:1957]:
```suggestion
    // XXX: Constrain the `stream` parameter using `@NotMatches(IsIdentityOperation.class)` once
    // `IsIdentityOperation` no longer matches nullary method invocations.
    @BeforeTemplate
```
Is this `XXX` also relevant in this case?

**Review by EnricSala — APPROVED** [review]
Thanks for picking up 💪

File: error-prone-contrib/src/test/resources/tech/picnic/errorprone/refasterrules/ReactorRulesTestOutput.java [hunk 2]
```diff
@@ -641,4 +643,8 @@ Mono<Void> testMonoFromFutureSupplier() {
   Mono<Void> testMonoFromFutureSupplierBoolean() {
     return Mono.fromFuture(() -> CompletableFuture.completedFuture(null), true);
   }
+
+  Flux<Integer> testFluxFromStreamSupplier() {
+    return Flux.fromStream(() -> RandomGenerator.StreamableGenerator.of(1));
```
Comment (Stephan202) [inline error-prone-contrib/src/test/resources/tech/picnic/errorprone/refasterrules/ReactorRulesTestOutput.java]: :eyes:

File: error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java (hunk 1 above)
Comment (Stephan202) [reply error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java:1967]: Yes, it is :+1:

File: error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java [hunk 3]
```diff
@@ -1913,7 +1914,7 @@ Duration after(StepVerifier.LastStep step, Duration duration) {
 
   /**
    * Prefer {@link Mono#fromFuture(Supplier)} over {@link Mono#fromFuture(CompletableFuture)}, as
-   * the former may defer initiation of the asynchornous computation until subscription.
+   * the former may defer initiation of the asynchronous computation until subscription.
```
Comment (Stephan202) [inline error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java:1924]: Nice one; should also be fixed below.

File: error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java [hunk 4]
```diff
@@ -1947,4 +1948,20 @@ Mono<T> after(CompletableFuture<T> future, boolean suppressCancel) {
...
   }
+
+  /**
+   * Prefer {@link Flux#fromStream(Supplier)} over {@link Flux#fromStream(Stream)}, as the former
+   * may defer initiation of the asynchronous computation until subscription.
+   */
+  static final class FluxFromStreamSupplier<T> {
+    @BeforeTemplate
+    Flux<T> before(Stream<T> stream) {
+      return Flux.fromStream(stream);
```
Comment (Stephan202) [inline error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java:1969]: This triggers an interaction with `FluxFromIterable`, so more changes are necessary; will apply.

File: error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java [hunk 5]
```diff
@@ -1947,4 +1948,20 @@ Mono<T> after(CompletableFuture<T> future, boolean suppressCancel) {
       return Mono.fromFuture(() -> future, suppressCancel);
     }
   }
+
+  /**
+   * Prefer {@link Flux#fromStream(Supplier)} over {@link Flux#fromStream(Stream)}, as the former
+   * may defer initiation of the asynchronous computation until subscription.
+   */
```
Comment (Stephan202) [inline error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java]: Not related to asynchronicity :eyes:.
```suggestion
  /**
   * Prefer {@link Flux#fromStream(Supplier)} over {@link Flux#fromStream(Stream)}, as the former
   * yields a {@link Flux} that is more likely to behave as expected when subscribed to more than
   * once.
   */
```

**Review by Stephan202 — APPROVED** [review]
Added a commit. Suggested commit message:
```
Introduce `FluxFromStreamSupplier` Refaster rule (#1261)
```

File: error-prone-contrib/src/test/resources/tech/picnic/errorprone/refasterrules/ReactorRulesTestOutput.java (hunk 2 above)
Comment (rickie) [reply error-prone-contrib/src/test/resources/tech/picnic/errorprone/refasterrules/ReactorRulesTestOutput.java]: Damn, I'm not sure what I was doing here 😬.

Comment (rickie) [reply error-prone-contrib/src/test/resources/tech/picnic/errorprone/refasterrules/ReactorRulesTestOutput.java]: I think something with auto-complete I'd say, but slipped through.

**Pull request merged**